File: plugin/src/main/java/io/j99/idea/vue/VueFileType.java
package io.j99.idea.vue;

import com.intellij.lang.xhtml.XHTMLLanguage;
import com.intellij.lang.xml.XMLLanguage;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * Created by apple on 16/2/26.
 */
public class VueFileType extends LanguageFileType {
    public static final VueFileType INSTANCE = new VueFileType();
    public static final String[] DEFAULT_ASSOCIATED_EXTENSIONS = {"vue"};

    protected VueFileType() {
        super(VueLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return VueBundle.message("vue.filetype.name");
    }

    @NotNull
    @Override
    public String getDescription() {
        return VueBundle.message("vue.filetype.description");
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return DEFAULT_ASSOCIATED_EXTENSIONS[0];
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return VueIcons.VUE_ICON;
    }
}

File: src/main/kotlin/io/j99/idea/vue/VueFileType.kt
package io.j99.idea.vue

import com.intellij.lang.html.HTMLLanguage
import com.intellij.openapi.fileTypes.LanguageFileType
import org.jetbrains.annotations.NonNls
import javax.swing.Icon

/**
 * Created by android on 15-12-25.
 */
class VueFileType : LanguageFileType(HTMLLanguage.INSTANCE) {
    override fun getName(): String {
        return VueBundle.message("vue.filetype.name")
    }

    override fun getDescription(): String {
        return VueBundle.message("vue.filetype.description")
    }

    override fun getDefaultExtension(): String {
        return DEFAULT_ASSOCIATED_EXTENSIONS[0]
    }

    override fun getIcon(): Icon? {
        return VueIcons.VUE_ICON
    }

    companion object {
        @JvmField val INSTANCE = VueFileType()
        @NonNls
        val DEFAULT_ASSOCIATED_EXTENSIONS = arrayOf("vue")
    }
}
